core/src/main/java/org/elasticsearch/index/query/MultiMatchQueryBuilder.java

```diff
@@ -747,7 +748,11 @@ protected Query doToQuery(QueryShardContext context) throws IOException {
 
         Query query = multiMatchQuery.parse(type, newFieldsBoosts, value, minimumShouldMatch);
         if (query == null) {
-            return null;
+            if (newFieldsBoosts.isEmpty()) {
```

Comment:
I was looking into this but parse() itself is quiet deep and delegates query creation to lots of other places that can potentially return null (haven't checked them all, though. Lots of the actual query creation can e.g. fieldType.fuzzyQuery() which in itselve has lots of implementations). So I though it best to catch any null return value here and convert it into MatchNoDocs().